controllers/socialUserController.js

```diff
@@ -0,0 +1,77 @@
+import models from '../models';
+
+export default class User {
+  static async userFacebookGoogle(req, res) {
+    // console.log('in controller =====', req.user);
+    try {
+      const facebookGoogleUser = await models.User.findOrCreate({
+        where: { email: req.user.emails[0].value },
+        defaults: {
+          firstName: req.user.name.givenName,
+          lastName: req.user.name.familyName,
+          username: req.user.username,
+          provider: req.user.provider
+        }
+      });
+      if (facebookGoogleUser) {
+        return res.status(201).json({
+          message: 'user successful registered',
+          user: {
+            // token,
+            firstName: req.user.name.givenName,
+            lastName: req.user.name.familyName,
+            email: req.user.emails[0].value,
+            provider: req.user.provider
+          }
+        });
+      }
+    } catch (error) {
+      return res.status(500).json({
```

Comment:
Improve on this error ro be a little specific by testing the errors encountered in the try and manifesting on them in the catch